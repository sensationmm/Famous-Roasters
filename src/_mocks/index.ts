import { CatalogueMockError, CatalogueMockMissingData, CatalogueMocks, CatalogueMocksPagination } from './CatalogueMock'
import { ProductMock } from './ProductMock'

export { CatalogueMocks as CatalogueMocks }
export { CatalogueMocksPagination as CatalogueMocksPagination }
export { CatalogueMockError as CatalogueMockError }
export { CatalogueMockMissingData as CatalogueMockMissingData }
export { ProductMock as ProductMock }
