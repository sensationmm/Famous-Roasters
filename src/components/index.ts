import { Button, ButtonColor, ButtonEmphasis, ButtonSize } from './Button'
import { Footer } from './Footer'
import { Layout } from './Layout'
import { Navigation } from './Navigation'
import { Typography, TypographySize, TypographyType } from './Typography'

export { Button as Button }
export { ButtonColor as ButtonColor }
export { ButtonEmphasis as ButtonEmphasis }
export { ButtonSize as ButtonSize }
export { Footer as Footer }
export { Layout as Layout }
export { Navigation as Navigation }
export { Typography as Typography }
export { TypographyType as TypographyType }
export { TypographySize as TypographySize }
