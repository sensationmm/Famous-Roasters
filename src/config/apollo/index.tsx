import { ApolloClient, ApolloLink, createHttpLink, from, InMemoryCache, NormalizedCacheObject } from '@apollo/client'
import { onError } from '@apollo/client/link/error'

export const apolloClient = (): ApolloClient<NormalizedCacheObject> => {
  const errorLink = onError(({ graphQLErrors, networkError }) => {
    if (graphQLErrors)
      graphQLErrors.forEach(({ message: errorMessage, locations, path }) => {
        console.log(`[GraphQL error]: Message: ${errorMessage}, Location: ${locations}, Path: ${path}`)
      })

    if (networkError) console.log(`[Network error]: ${networkError}`)
  })

  return new ApolloClient({
    cache: new InMemoryCache(),
    link: from([
      errorLink,
      createHttpLink({
        uri: process.env.REACT_APP_SHOPIFY_STOREFRONT_GRAPHQL_ENDPOINT,
        headers: {
          'X-Shopify-Storefront-Access-Token': process.env.REACT_APP_SHOPIFY_STOREFRONT_ACCESS_TOKEN,
          Accept: 'application/graphql',
        },
        fetch,
      }) as unknown as ApolloLink,
    ]),
  })
}
