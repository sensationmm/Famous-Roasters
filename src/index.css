@tailwind base;
@tailwind components;
@tailwind utilities;

body {
  margin: 0;
  padding: 0;
  font-family: 'Open Sans', sans-serif;
  font-size: 16px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

body > div > div > main {
  min-height: 50vh;
}

:root {
  --swiper-theme-color: #646575 !important;
}

.video-container {
  position: relative;
  padding-bottom: 56.25%;
  padding-top: 0;
  height: 0;
  overflow: hidden;
}

.video-container iframe,
.video-container object,
.video-container embed {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}