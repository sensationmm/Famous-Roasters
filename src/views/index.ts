import { About } from './About'
import { Account } from './Account'
import { Cart } from './Cart'
import { Catalogue } from './Catalogue'
import { Checkout } from './Checkout'
import { Contact } from './Contact'
import { Error } from './Error'
import { Home } from './Home'
import { OurRoasters } from './OurRoasters'
import { Product } from './Product'
import { TasteFinder } from './TasteFinder'

export { About as About }
export { Account as Account }
export { Cart as Cart }
export { Catalogue as Catalogue }
export { Checkout as Checkout }
export { Contact as Contact }
export { Error as Error }
export { Home as Home }
export { OurRoasters as OurRoasters }
export { Product as Product }
export { TasteFinder as TasteFinder }
