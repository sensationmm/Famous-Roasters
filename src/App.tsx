import React from 'react'
import { I18nextProvider } from 'react-i18next'
import { BrowserRouter, Route, Routes } from 'react-router-dom'
import {
  About,
  Account,
  Cart,
  Catalogue,
  Checkout,
  Contact,
  Error,
  Home,
  OurRoasters,
  Product,
  TasteFinder,
} from 'src/views'

import { i18n } from './config'

function App() {
  return (
    <I18nextProvider i18n={i18n}>
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/about" element={<About />} />
          <Route path="/account" element={<Account />} />
          <Route path="/cart" element={<Cart />} />
          <Route path="/catalogue" element={<Catalogue />} />
          <Route path="/checkout" element={<Checkout />} />
          <Route path="/contact" element={<Contact />} />
          <Route path="/our-roasters" element={<OurRoasters />} />
          <Route path="/product/:id" element={<Product />} />
          <Route path="/taste-finder" element={<TasteFinder />} />
          <Route path="*" element={<Error />} />
        </Routes>
      </BrowserRouter>
    </I18nextProvider>
  )
}

export default App
