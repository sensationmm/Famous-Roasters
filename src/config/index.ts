import { apolloClient } from './apollo'
import i18n from './i18n'

export { i18n as i18n }
export { apolloClient as apolloClient }
