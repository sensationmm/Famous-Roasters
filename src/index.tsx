import './index.css'

import React from 'react'
import { createRoot } from 'react-dom/client'

import App from './App'

const container = document.getElementById('root') as Element
const root = createRoot(container)
if (process.env.NODE_ENV === 'development') {
  root.render(
    <React.StrictMode>
      <App />
    </React.StrictMode>,
  )
} else {
  root.render(<App />)
}
