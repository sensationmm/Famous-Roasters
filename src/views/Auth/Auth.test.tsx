import { render, waitFor } from '@testing-library/react'
import Amplify from 'aws-amplify'
import React from 'react'
import { I18nextProvider } from 'react-i18next'
import { MemoryRouter } from 'react-router-dom'
import { i18n } from 'src/config'
import { awsconfig } from 'src/config/cognito/auth.hook'

import { Auth } from '.'

Amplify.configure(awsconfig)

describe('Auth view', () => {
  it('Renders correctly', async () => {
    const { container } = render(
      <I18nextProvider i18n={i18n}>
        <MemoryRouter initialEntries={['/login']}>
          <Auth authState={'signIn'} />
        </MemoryRouter>
      </I18nextProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })
})
