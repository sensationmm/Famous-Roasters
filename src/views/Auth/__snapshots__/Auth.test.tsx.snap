// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth view Renders correctly 1`] = `
<div>
  <div
    class="min-h-screen flex flex-col text-brand-black"
  >
    <div
      class="bg-white z-10"
    >
      <header
        class="relative bg-brand-black"
      >
        <nav
          aria-label="Top"
          class="max-w-7xl mx-auto px-4 md:px-6 xl:px-8"
        >
          <div
            class="h-16 flex items-center justify-between border-b border-brand-black"
          >
            <div
              class="flex-1 flex items-center xl:hidden"
            >
              <button
                class="-ml-2 bg-brand-black p-2 rounded-md text-gray-400"
                data-testid="button-open"
                type="button"
              >
                <span
                  class="sr-only"
                >
                  Open menu
                </span>
                <svg
                  class="undefined fill-brand-black stroke-brand-black"
                  fill="inherit"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    clip-rule="evenodd"
                    d="M3.17188 5H20.0156C20.1102 5 20.1875 5.07734 20.1875 5.17188V6.375C20.1875 6.46953 20.1102 6.54688 20.0156 6.54688H3.17188C3.07734 6.54688 3 6.46953 3 6.375V5.17188C3 5.07734 3.07734 5 3.17188 5ZM3.17188 11.5469H20.0156C20.1102 11.5469 20.1875 11.6242 20.1875 11.7188V12.9219C20.1875 13.0164 20.1102 13.0938 20.0156 13.0938H3.17188C3.07734 13.0938 3 13.0164 3 12.9219V11.7188C3 11.6242 3.07734 11.5469 3.17188 11.5469ZM20.0156 18.0938H3.17188C3.07734 18.0938 3 18.1711 3 18.2656V19.4688C3 19.5633 3.07734 19.6406 3.17188 19.6406H20.0156C20.1102 19.6406 20.1875 19.5633 20.1875 19.4688V18.2656C20.1875 18.1711 20.1102 18.0938 20.0156 18.0938Z"
                    fill="#FFFFFF"
                    fill-rule="evenodd"
                    stroke="#FFFFFF"
                    stroke-width="0.25"
                  />
                </svg>
              </button>
            </div>
            <div
              class="hidden xl:flex-1 xl:block xl:self-stretch"
            >
              <div
                class="h-full flex space-x-10"
              >
                <a
                  class="flex items-center text-white hover:text-coreUI-text-tertiary pb-1.5 duration-500"
                  href="//www.60beans.com/uber-uns"
                >
                  <span
                    class="uppercase text-sm leading-5"
                  >
                    Über uns
                  </span>
                </a>
                <a
                  class="flex items-center text-white hover:text-coreUI-text-tertiary pb-1.5 duration-500"
                  href="//www.60beans.com/roesterei"
                >
                  <span
                    class="uppercase text-sm leading-5"
                  >
                    Unsere Röster
                  </span>
                </a>
                <a
                  class="flex items-center text-white hover:text-coreUI-text-tertiary pb-1.5 duration-500"
                  href="/"
                >
                  <span
                    class="uppercase text-sm leading-5"
                  >
                    Blog
                  </span>
                </a>
              </div>
            </div>
            <a
              class="flex h-full items-center pl-2 pr-2"
              href="//www.60beans.com"
            >
              <span
                class="sr-only"
              >
                60beans
              </span>
              <img
                alt="60beans"
                class="h-6"
                src="60beans-light.svg"
              />
            </a>
            <div
              class="flex-1 flow-root flex h-full items-center"
            >
              <div
                class="flex-1 flex h-full items-center justify-end xl:flex-none"
              >
                <div
                  class="hidden h-full xl:flex space-x-8 justify-end"
                >
                  <a
                    class="flex items-center text-white hover:text-coreUI-text-tertiary pb-1.5 duration-500"
                    href="/taste-finder"
                  >
                    <span
                      class="uppercase text-sm leading-5"
                    >
                      Tastefinder
                    </span>
                  </a>
                </div>
                <a
                  class="hidden xl:block ml-10"
                  href="/profile"
                >
                  <svg
                    class="undefined fill-brand-black stroke-brand-black"
                    fill="inherit"
                    height="24"
                    viewBox="0 0 24 25"
                    width="24"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M16 8.66663C16 10.8766 14.21 12.6666 12 12.6666C9.79 12.6666 8 10.8766 8 8.66663C8 6.45663 9.79 4.66663 12 4.66663C14.21 4.66663 16 6.45663 16 8.66663ZM4 18.6666C4 16.0066 9.33 14.6666 12 14.6666C14.67 14.6666 20 16.0066 20 18.6666V20.6666H4V18.6666Z"
                      fill="#FFFFFF"
                      fill-rule="evenodd"
                      stroke="none"
                      stroke-width="none"
                    />
                  </svg>
                </a>
                <div
                  class="ml-4 flow-root xl:ml-10"
                >
                  <button
                    class="relative inline-flex flex-row items-center rounded-full border-2 px-3 py-2 text-sm font-semibold disabled:bg-brand-grey-whisper disabled:text-coreUI-text-secondary disabled:cursor-not-allowed text-brand-black bg-primary border-transparent hover:bg-brand-green-starship active:bg-primary"
                    data-testid="button-shop"
                    role="button"
                  >
                    Shop
                  </button>
                </div>
              </div>
            </div>
          </div>
        </nav>
      </header>
    </div>
    <main
      class="flex-grow flex items-start justify-center bg-white"
    >
      <div
        class="w-full md:w-80 xl:w-auto xl:w-[32rem]"
      >
        <div
          class="Section__container___3YYTG"
        />
      </div>
    </main>
  </div>
</div>
`;
